### PR TITLE
fix: handle multiple masters for Azure Stack

### LINKED ISSUE #991: SetPropertiesDefaults errors if masterProfile.count > 1
**Is this an ISSUE or FEATURE REQUEST?** (choose one):

ISSUE

---

**What version of aks-engine?**:

Current master, post v0.33.5.

---

I noticed after #947 merged that we weren't checking the error return from `SetPropertiesDefaults` in some tests. After changing the code to handle it, the test kubernetes.json apimodel for azurestack throws this error unless `masterProfile.count` is set to 1:
```
The raw pem is not a valid PEM formatted block:
caCertificate
```

cc: @honcao

### PR DESCRIPTION
**Pull request by honcao**
**Reason for Change**:

Azure Stack can not use Internal load balancer to set up the K8s cluster. Azure Stack team is working on the fix and it will be release in next release. currently, the PR route the traffic to public load balancer.

**Issue Fixed**:

Fixes #991

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by honcao** [comment]
FYI @deaborch @jadarsie @rjaini @vikasnav @Bhuvaneswari-Santharam

**Comment by mboersma** [comment]
I think this will fix #991. @honcao you might consider catching the test errors from `SetPropertiesDefaults` to validate this fix, or we can wait and I will merge #992 right after.

File: parts/k8s/cloud-init/masternodecustomdata.yml [hunk 1]
```diff
@@ -334,6 +334,16 @@ MASTER_CONTAINER_ADDONS_PLACEHOLDER
 {{ else }}
     sed -i "s|<etcdEndPointUri>|127.0.0.1|g" $a
 {{ end }}
+{{if IsAzureStackCloud}}
+    {{if IsMultipleMasters}}
+    masterLBIP=`getent hosts  {{WrapAsVariable "masterPublicLbFQDN"}} |  cut -d " " -f1`
```
Comment (mboersma) [inline parts/k8s/cloud-init/masternodecustomdata.yml]: Some extra spaces here, after `hosts` and the vertical bar `|`.

File: pkg/engine/armvariables.go [hunk 2]
```diff
@@ -157,6 +158,7 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		}
 		masterVars["environmentJSON"] = environmentJSON
 		masterVars["provisionConfigsCustomCloud"] = getBase64EncodedGzippedCustomScript(kubernetesCSECustomCloud)
+		masterVars["masterPublicLbFQDN"] = fmt.Sprintf("%s.%s.%s", strings.ToLower(cs.Properties.MasterProfile.DNSPrefix), strings.ToLower(cs.Location), strings.ToLower(cs.GetCloudSpecConfig().EndpointConfig.ResourceManagerVMDNSSuffix))
```
Comment (mboersma) [inline pkg/engine/armvariables.go]: Could you split this onto multiple lines?

File: parts/k8s/cloud-init/nodecustomdata.yml [hunk 3]
```diff
@@ -227,7 +227,15 @@ write_files:
     - name: localcluster
       cluster:
         certificate-authority: /etc/kubernetes/certs/ca.crt
+        {{if IsAzureStackCloud}}
```
Comment (rjaini) [inline parts/k8s/cloud-init/nodecustomdata.yml]: Should we not needed to change dcosmasterresources.t file to make publicIP Allocation as static. https://github.com/msazurestackworkloads/aks-engine/pull/2/files

File: parts/k8s/cloud-init/nodecustomdata.yml [hunk 4]
```diff
@@ -227,7 +227,15 @@ write_files:
...
+        {{if IsMultipleMasters}}
+        server: https://<kubernetesAPIServerIP>:443
```
Comment (rjaini) [inline parts/k8s/cloud-init/nodecustomdata.yml]: Shouldn't this wrap to masterPublicLbFQDN ?.

File: parts/k8s/cloud-init/nodecustomdata.yml (hunk 3 above)
Comment (mboersma) [reply parts/k8s/cloud-init/nodecustomdata.yml]: DC/OS support is deprecated, so we really want to leave those files frozen as-is unless there's a compelling reason.

Comment (honcao) [reply parts/k8s/cloud-init/nodecustomdata.yml]: in upstream, it is already static right now.

**Pull request merged**